plugins/ipam/dhcp/lease.go

```diff
@@ -72,29 +72,34 @@ func AcquireLease(clientID, netns, ifName string) (*DHCPLease, error) {
 	log.Printf("%v: acquiring lease", clientID)
 
 	l.wg.Add(1)
-	go ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
-		defer l.wg.Done()
-
-		link, err := netlink.LinkByName(ifName)
-		if err != nil {
-			errCh <- fmt.Errorf("error looking up %q", ifName)
-			return
-		}
+	go func() {
+		err := ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
+			defer l.wg.Done()
+
+			link, err := netlink.LinkByName(ifName)
+			if err != nil {
+				errCh <- fmt.Errorf("error looking up %q", ifName)
+				return
+			}
 
-		l.link = link
+			l.link = link
 
-		if err = l.acquire(); err != nil {
-			errCh <- err
-			return
-		}
+			if err = l.acquire(); err != nil {
+				errCh <- err
```

Comment:
The closure returns err (!= nil), then the goroutine also pushes that error into the ch on line 100 right?